cmd/root.go

```diff
@@ -215,15 +216,17 @@ func printCheckResults(enabledChecks checker.CheckNameToFnMap) {
 }
 
 // makeRepo helps turn a URI into the appropriate clients.Repo.
-// currently this is a decision between GitHub and GitLab,
+// currently this is a decision between GitHub, GitLab, and Azure DevOps,
 // but may expand in the future.
 func makeRepo(uri string) (clients.Repo, error) {
 	var repo clients.Repo
-	var errGitHub, errGitLab error
+	var errGitHub, errGitLab, errAzureDevOps error
 	if repo, errGitHub = githubrepo.MakeGithubRepo(uri); errGitHub != nil {
-		repo, errGitLab = gitlabrepo.MakeGitlabRepo(uri)
-		if errGitLab != nil {
-			return nil, fmt.Errorf("unable to parse as github or gitlab: %w", errors.Join(errGitHub, errGitLab))
+		if repo, errGitLab = gitlabrepo.MakeGitlabRepo(uri); errGitLab != nil {
+			repo, errAzureDevOps = azuredevopsrepo.MakeAzureDevOpsRepo(uri)
+			if errAzureDevOps != nil {
```

Comment:
experimental gate here?

Your other gate was in `checker.GetClients`, which isn't used by most of the code